src/lammpsio/snapshot.py

```diff
@@ -35,19 +36,25 @@ class Snapshot:
 
     """
 
-    def __init__(self, N, box, step=None):
+    def __init__(self, N, box, num_types, step=None):
         self._N = N
         self._box = Box.cast(box)
+        self.num_types = num_types
         self.step = step
 
         self._id = None
         self._position = None
         self._velocity = None
         self._image = None
         self._molecule = None
+        self._num_types = None
         self._typeid = None
         self._charge = None
         self._mass = None
+        self._bonds = None
+        self._angles = None
+        self._dihedrals = None
+        self._impropers = None
 
     @classmethod
     def from_hoomd_gsd(cls, frame):
```

Comment:
Making a note that we need to come back here and support the conversion of bonds to/from GSD. You can decide if that is this PR or a future one.